src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,82 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin("abcdef@gmail.com");
+        user.setAge(22);
+        user.setPassword("abc123");
+
+    }
+
+    @Test
+    void register_sameUserLogin_NotOk() {
+        Storage.people.add(user);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_nullLogin_NotOk() {
+        user.setLogin(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void expectedBehaviorValid_SetLogin_Ok() {
+        final User actual = registrationService.register(user);

```

Comment:
Why final?

Reply:
In Intellij IDEA inserted it automatically, I did not delete it